credentials/src/main/kotlin/web5/sdk/credentials/StatusListCredential.kt

```diff
@@ -0,0 +1,317 @@
+
+/**
+ * The JSON property key for an encoded list.
+ */
+private const val ENCODED_LIST: String = "encodedList"
+
+/**
+ * The JSON property key for a status purpose.
+ */
+private const val STATUS_PURPOSE: String = "statusPurpose"
+
+/**
+ * `StatusListCredential` represents a digitally verifiable status list credential according to the
+ * [W3C Verifiable Credentials Status List v2021](https://www.w3.org/TR/vc-status-list/).
+ *
+ * When a status list is published, the result is a verifiable credential that encapsulates the status list.
+ *
+ */
+public object StatusListCredential {
+  /**
+   * Create a [StatusListCredential] with a specific purpose, e.g., for revocation.
+   *
+   * @param statusListCredentialId The id used for the resolvable path to the status list credential [String].
+   * @param issuer The issuer URI of the credential, as a [String].
+   * @param statusPurpose The status purpose of the status list cred, eg: revocation, as a [StatusPurpose].
+   * @param issuedCredentials The credentials to be included in the status list credential, eg: revoked credentials, list of type [VerifiableCredential].
+   * @return A [VerifiableCredential] instance.
+   *
+   * Example:
+   * ```
+   * val statusListCredential = StatusListCredential.create("http://example.com/statuslistcred/id123", "http://example.com/issuers/1", StatusPurpose.REVOCATION, listOf(vc1,vc2))
+   * ```
+   */
+  @Throws(RuntimeException::class)
+  public fun create(
+    statusListCredentialId: String,
+    issuer: String,
+    statusPurpose: StatusPurpose,
+    issuedCredentials: List<VerifiableCredential>
+  ): VerifiableCredential {
+    val statusListIndexes: List<String>
+    val bitString: String
+
+    try {
+      statusListIndexes = prepareCredentialsForStatusList(statusPurpose, issuedCredentials)
+      bitString = bitstringGeneration(statusListIndexes)
+    } catch (e: Exception) {
+      throw RuntimeException("An error occurred during the creation of the status list credential: ${e.message}", e)
+    }
+
+    val claims = mapOf(STATUS_PURPOSE to statusPurpose.toString().lowercase(), ENCODED_LIST to bitString)
+    val credSubject = CredentialSubject.builder()
+      .id(URI.create(statusListCredentialId))
+      .type("StatusList2021")
+      .claims(claims)
+      .build()
+
+    val vcDataModel = VcDataModel.builder()
+      .id(URI.create(statusListCredentialId))
+      .issuer(URI.create(issuer))
+      .issuanceDate(Date())
+      .context(URI.create("https://w3id.org/vc/status-list/2021/v1"))
+      .type("StatusList2021Credential")
+      .credentialSubject(credSubject)
+      .build()
+
+    return VerifiableCredential(vcDataModel)
+  }
+
+  /**
+   * Validates if a given credential is part of the status list represented by a [VerifiableCredential].
+   *
+   * @param credentialToValidate The [VerifiableCredential] to be validated against the status list.
+   * @param statusListCredential The [VerifiableCredential] representing the status list.
```

Comment:
Curious why you decided to pass in both credentials? An alternative can be to pass in the `credentialToValidate` only, and have this method retrieve the status list. What was the rationale to pick one vs the other?

Reply:
I did both because the issuer may have these objects not hosted yet.

```
  public fun validateCredentialInStatusList(
    credentialToValidate: VerifiableCredential,
    statusListCredential: VerifiableCredential
  ): Boolean {
  ```
  
  ```
    public suspend fun validateCredentialInStatusList(
    credentialToValidate: VerifiableCredential,
    httpClient: HttpClient? = null // default HTTP client but can be overridden
  ): Boolean {
  ```